source/material_model/utilities.cc

```diff
@@ -699,6 +699,43 @@ namespace aspect
       }
 
 
+      // Calculate a parameter of single composition at single point by averaging phases.
+      // Either linear(by_log is 0) or using log form(by_log is 1), default is linear.
+      double phase_average_value (const std::pair<std::vector<double>, const std::vector<unsigned int>> &gamma_inputs,
+                                  const std::vector<double> &parameter_values,
+                                  const unsigned int composition,
+                                  bool by_log)
+      {
+        const std::vector<double> &gamma_values = gamma_inputs.first;
+        const std::vector<unsigned int> &n_phases_per_composition = gamma_inputs.second;
+        // Calculate base index and assign base value
+        unsigned int base = 0;
+        for (unsigned int i=0; i<composition; ++i)
+          base += n_phases_per_composition[i] + 1;
+        double averaged_parameter = parameter_values[base];
+        if (n_phases_per_composition[composition] > 0)
+          {
+            // Do averaging when there are multiple phases
+            if (by_log)
+              averaged_parameter = log(averaged_parameter);
+            for (unsigned int i=0; i<n_phases_per_composition[composition]; ++i)
```

Comment:
add a newline before the `for` to make clear that line 720 is the only line inside the `if (log)`.